diff --git a/lib/server/chromium/crExecutionContext.js b/lib/server/chromium/crExecutionContext.js
--- a/lib/server/chromium/crExecutionContext.js
+++ b/lib/server/chromium/crExecutionContext.js
@@ -104,6 +104,8 @@ function rewriteError(error) {
     throw new Error("Cannot serialize result: object reference chain is too long.");
   if (error.message.includes("Object couldn't be returned by value"))
     return { result: { type: "undefined" } };
+  if (error.message.includes("Promise was collected"))
+    throw new Error("Resulting promise was garbage collected.");
   if (error instanceof TypeError && error.message.startsWith("Converting circular structure to JSON"))
     (0, import_stackTrace.rewriteErrorMessage)(error, error.message + " Are you passing a nested JSHandle?");
   if (!js.isJavaScriptErrorInEvaluate(error) && !(0, import_protocolError.isSessionClosedError)(error))
